BlueprintUI/Sources/Element/EnvironmentAdaptingStorage.swift

```diff
@@ -10,12 +13,19 @@ struct EnvironmentAdaptingStorage: ContentStorage {
     var adapter: (inout Environment) -> Void
 
     var child: Element
+    var content: ElementContent
```

Comment:
So @kdaker has a good demo of how this can fail (can you link to your branch? I can't find it), but it's not so much about grabbing the content of the GeometryReader itself, but what _it_ returns. Again, I'm not 100% sure if this happens here, but I'll describe the issue we saw:

1) A `GeometryReader` which forks behavior based on layout size (you can imagine other similar cases), eg:
```swift
GeometryReader { geometry in
   if geometry.width < 200 {
      return Label(text: "A shorter string")
   } else {
      return Label(text: "A much longer string that is longer")
   }
}
```

2) The cache caches the first fetch of the label & its content under the `ElementIdentifier` `Label.1`. This is usually the `else` case in the above during measurement.

3) Later fetches / accesses do not update the underlying `content`, meaning if the final layout hits the primary `if` case, the wrong content is returned.

Reply:
Here's the code he was using: 

```swift
final class PlaygroundViewController: UIViewController {
    let blueprintView = BlueprintView()
    
    lazy var model: [Element] = [geometryElement, geometryElement, geometryElement]
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.view = blueprintView
        
        update()
    }
    
    var geometryElement: Element {
        GeometryReader { proxy in
            return Row { row in
                if proxy.constraint.width.maximum >= 60 {
                    row.addFixed(child: Label(text: String("Big label")).box(background: .blue))
                } else {
                    row.addFixed(child: Label(text: String("Small label")).box(background: .green))
                }
                
            }
        }
    }
    
    func update() {
        blueprintView.element = element(for: model)
    }
    
    func element(for model: [Element]) -> Element {
        Row { row in
            for item in self.model {
                let itemElement = Box(
                    backgroundColor: .red,
                    wrapping: item
                ).tappable {
                    self.model.append(self.geometryElement)
                    self.update()
                }
                row.add(child: itemElement)
            }
        }
        .constrainedTo(size: .init(width: 400, height: 300))
        .centered()
    }
}
```

Reply:
I see what you're saying. I don't think the `content` stored here will cause a problem, but we could have a similar issue with the `LayoutTreeNode` that gets passed to the lazily created child, since it can have cached things hanging off it (i.e. `LayoutSubelement`). I'll set up a regression test like this to make sure we're not breaking anything.